src/api/ApiProvider.js

```diff
@@ -0,0 +1,304 @@
+import {
+    collection,
+    deleteDoc,
+    doc,
+    getDoc,
+    getDocs,
+    onSnapshot,
+    query,
+    setDoc,
+    where
+} from 'firebase/firestore';
+import { db } from './Firebase';
+
+class ApiProvider {
```

Comment:
This should be I think ApiService, for providing data, we should actually create a custom hook, which based on arguments will return us the data object and loading state (errors handled with react-toastify ofc)